### PR TITLE
Assessments/unit 01 booleans and strings

### PR DESCRIPTION
**Pull request by EllisWaterman**

### CODE REVIEW

**Review by gigamonkey — APPROVED** [review]
Pretty good job. You can go ahead and merge this. If you want to fix any of the ones you got wrong or take a crack at the last one, you can submit another PR and I'll be happy to review your changes. (It'll be easier to do that if you merge this first though.)

File: assessments/unit-01-booleans-and-strings/code.js [hunk 1]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
 // syntactically correct function skeleton, more credit for a skeleton with a
 // reasonable argument list, yet more credit for a working function, and full
 // credit for a clear and simple working function.
+
+const fireAlarm = (fireAlarmPulled, smokeDetected, plannedDrill) => {
+  return fireAlarmPulled || smokeDetected || plannedDrill
+}
+
+const canBePresident = (is35, naturalBornCitizen, livedInUsFor14Years) => {
+  return is35 && naturalBornCitizen && livedInUsFor14Years
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js:20]: Fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 2]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+}
+
+const willSeeTweet = (followTweeter, followedSomoneWhoReTweeted, blockedOGTweeter) => {
+  return !blockedOGTweeter || followTweeter || followedSomoneWhoReTweeted
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js]: Fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 3]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+const evenGreaterThanZero = (x) => {
+  if (x > 0)
+    if (x % 2 === 0)
+      return true
+    else
+      if (x < 0)
+        return false
+}
+
+const isLeapYear = (year) => {
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js:33]: fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 4]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+  if (year % 100 === 0)
+    return false
+  if (year % 4 === 0)
+    return true
+  else
+    return false
+}
+
+const firstAndLast = (s) => {
+  if (s.length > 1)
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js]: fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 5]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+  return s.substring(posOfFirstVowel) + s.substring(0, posOfFirstVowel) + 'ay'
+}
+
+const randomCharacter = (s) => {
+  return s.substring(rand(s.length), rand(s.length+1))
+  //unfinished
+}
+
+const randomCharacterUpDown = (s) => {
+  if (rand())
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js]: fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 6]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+  return s.substring(rand(s.length), rand(s.length+1)).toUpperCase()
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js]: fixed

File: assessments/unit-01-booleans-and-strings/code.js [hunk 7]
```diff
@@ -9,3 +9,71 @@ const rand = (n) => Math.floor(Math.random() * n);
...
+  else
+  return s.substring(rand(s.length), rand(s.length+1)).toLowerCase()
+  //unfinished
+}
+
+const isAllUpperCase = (s) => {
+  if (s.substring(0) == s.substring(0).toUpperCase())
```
[the remark this answers is not in the document]
Comment (EllisWaterman) [reply assessments/unit-01-booleans-and-strings/code.js]: fixed

**Comment by EllisWaterman** [comment]
Last one seemed easy, ran out of time.

**Pull request merged**